src/cpp/client.cpp

```diff
@@ -30,386 +30,408 @@
-        tensor = new Tensor<int8_t>(g_key, (void*)blob.data(),
-                                    reply_dims, reply_type,
-                                    MemoryLayout::contiguous);
-        break;
-        case TensorType::uint16 :
-        tensor = new Tensor<uint16_t>(g_key, (void*)blob.data(),
-                                      reply_dims, reply_type,
-                                      MemoryLayout::contiguous);
-        break;
-        case TensorType::uint8 :
-        tensor = new Tensor<uint8_t>(g_key, (void*)blob.data(),
-                                     reply_dims, reply_type,
-                                     MemoryLayout::contiguous);
-        break;
+    // Retrieve the tensor data into a Tensor
+    std::string_view blob = GetTensorCommand::get_data_blob(reply);
+    TensorBase* tensor = NULL;
+    switch (reply_type) {
+        case TensorType::dbl:
+            tensor = new Tensor<double>(get_key, (void*)blob.data(),
+                                        reply_dims, reply_type,
+                                        MemoryLayout::contiguous);
+            break;
+        case TensorType::flt:
+            tensor = new Tensor<float>(get_key, (void*)blob.data(),
+                                       reply_dims, reply_type,
+                                       MemoryLayout::contiguous);
+            break;
+        case TensorType::int64:
+            tensor = new Tensor<int64_t>(get_key, (void*)blob.data(),
+                                         reply_dims, reply_type,
+                                         MemoryLayout::contiguous);
+            break;
+        case TensorType::int32:
+            tensor = new Tensor<int32_t>(get_key, (void*)blob.data(),
+                                         reply_dims, reply_type,
+                                         MemoryLayout::contiguous);
+            break;
+        case TensorType::int16:
+            tensor = new Tensor<int16_t>(get_key, (void*)blob.data(),
+                                         reply_dims, reply_type,
+                                         MemoryLayout::contiguous);
+            break;
+        case TensorType::int8:
+            tensor = new Tensor<int8_t>(get_key, (void*)blob.data(),
+                                        reply_dims, reply_type,
+                                        MemoryLayout::contiguous);
+            break;
+        case TensorType::uint16:
+            tensor = new Tensor<uint16_t>(get_key, (void*)blob.data(),
+                                          reply_dims, reply_type,
+                                          MemoryLayout::contiguous);
+            break;
+        case TensorType::uint8:
+            tensor = new Tensor<uint8_t>(get_key, (void*)blob.data(),
+                                         reply_dims, reply_type,
+                                         MemoryLayout::contiguous);
+            break;
         default : // FINDME: Need to handle this case better
```

Comment:
Just commenting here for bookkeeping.  If we need to add the previous ``FINDME`` as an issue, this one seems to match with it and could maybe be combined.